extensions/wikia/UserProfilePageV3/js/UserProfilePage.js

```diff
@@ -388,7 +389,10 @@ var UserProfilePage = {
 			type: 'POST',
 			url: this.ajaxEntryPoint + '&method=saveUserData',
 			dataType: 'json',
-			data: {'userId': UserProfilePage.userId, 'data': JSON.stringify(userData)},
+			data: {
+				'userId': UserProfilePage.userId,
+				'data': JSON.stringify(userData),
+				'token': token},
```

Comment:
You can simply call `window.mw.user.tokens.get('editToken')`here to get the edit token instead of passing it from the backend.